src/deployment/registration.py

```diff
@@ -10,78 +10,94 @@
 import urllib.parse
 from datetime import datetime, timedelta
 from enum import Enum
-from typing import Any, Callable, Dict, List, NamedTuple, Optional, Tuple, TypeVar
-from uuid import UUID, uuid4
+from typing import Any, Callable, Dict, List, NamedTuple, Optional, Tuple, TypeVar, cast
+from uuid import UUID
 
 import requests
-from azure.cli.core.azclierror import AuthenticationError
-from azure.common.client_factory import get_client_from_cli_profile
 from azure.common.credentials import get_cli_profile
-from azure.graphrbac import GraphRbacManagementClient
-from azure.graphrbac.models import (
-    Application,
-    ApplicationCreateParameters,
-    ApplicationUpdateParameters,
-    AppRole,
-    PasswordCredential,
-    RequiredResourceAccess,
-    ResourceAccess,
-    ServicePrincipal,
-    ServicePrincipalCreateParameters,
-)
 from functional import seq
 from msrest.serialization import TZ_UTC
 
-FIX_URL = (
-    "https://ms.portal.azure.com/#blade/Microsoft_AAD_RegisteredApps/"
-    "ApplicationMenuBlade/ProtectAnAPI/appId/%s/isMSAApp/"
-)
-
 logger = logging.getLogger("deploy")
 
+## https://docs.microsoft.com/en-us/graph/api/overview?view=graph-rest-1.0
+GRAPH_RESOURCE = "https://graph.microsoft.com"
+GRAPH_RESOURCE_ENDPOINT = "https://graph.microsoft.com/v1.0"
+
 
 class GraphQueryError(Exception):
-    def __init__(self, message: str, status_code: int) -> None:
+    def __init__(self, message: str, status_code: Optional[int]) -> None:
         super(GraphQueryError, self).__init__(message)
         self.message = message
         self.status_code = status_code
 
 
+## Queries microsoft graph api and return
 def query_microsoft_graph(
     method: str,
     resource: str,
     params: Optional[Dict] = None,
     body: Optional[Dict] = None,
-) -> Any:
+    subscription: Optional[str] = None,
+) -> Dict:
     profile = get_cli_profile()
     (token_type, access_token, _), _, _ = profile.get_raw_token(
-        resource="https://graph.microsoft.com"
+        resource=GRAPH_RESOURCE, subscription=subscription
     )
-    url = urllib.parse.urljoin("https://graph.microsoft.com/v1.0/", resource)
+    url = urllib.parse.urljoin(f"{GRAPH_RESOURCE_ENDPOINT}/", resource)
     headers = {
         "Authorization": "%s %s" % (token_type, access_token),
         "Content-Type": "application/json",
     }
     response = requests.request(
         method=method, url=url, headers=headers, params=params, json=body
     )
-
-    response.status_code
-
     if 200 <= response.status_code < 300:
-        try:
-            return response.json()
-        except ValueError:
-            return None
+        if response.content and response.content.strip():
+            return cast(Dict, response.json())
```

Comment:
Since a JSON document can also be an array of values, let's add a final `isinstance` check here, too.

So something like:
```suggestion
            json = response.json()
            if isinstance(json, dict):
                return cast(Dict, json)
            else:
                # return error
```

Reply:
(if we do the `isinstance` checks, I think we don't need to cast anymore, too)